### PR TITLE
8280366: (fs) Restore Files.createTempFile javadoc

### PR DESCRIPTION
**Pull request by bplb**
This change reverts commit fb8fdc0fbf17dd7e900cb688df4917b97b26b9ab and then makes a smaller change to the verbiage of `Files.createTempFile `.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8280366](https://bugs.openjdk.java.net/browse/JDK-8280366): (fs) Restore Files.createTempFile javadoc

### Reviewers
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7202/head:pull/7202` \
`$ git checkout pull/7202`

Update a local copy of the PR: \
`$ git checkout pull/7202` \
`$ git pull https://git.openjdk.java.net/jdk pull/7202/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7202`

View PR using the GUI difftool: \
`$ git pr show -t 7202`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7202.diff">https://git.openjdk.java.net/jdk/pull/7202.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/nio/file/Files.java [hunk 1]
```diff
@@ -823,11 +823,13 @@ private static void createAndCheckIsDirectory(Path dir,
      * names in the same manner as the {@link
      * java.io.File#createTempFile(String,String,File)} method.
      *
-     * <p> The file may be opened using the {@link
+     * <p> As with the {@code File.createTempFile} methods, this method is only
+     * part of a temporary-file facility. Where the resulting file is used
+     * as a <em>work file</em> solely by a single instance of the Java virtual
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/nio/file/Files.java:828]: I think the insertion of "solely by a single instance of the Java virtual machine" will create questions and we'll end up re-writing the paragraph again. I think I'd prefer to just restore the text and then start again with the questions that lead to the change in the first place.

**Review by AlanBateman — APPROVED** [review]

**Review by LanceAndersen — APPROVED** [review]
This is Looks fine to get us back to where we started.  I do think we need to revisit some of the questions that were asked about the original wording via a new PR

**Comment by bplb** [comment]
Specifically, there is no definition of either _temporary-file facility_ or _work file_.

**Pull request closed without merging**